### PR TITLE
[IR] Avoid duplicate IR elements in for loops lowering.

### PR DESCRIPTION
**Pull request by madsager**

### CODE REVIEW

**Comment by madsager** [comment]
@max-kammerer @homuroll 

This should fix the duplicate IR elements noticed when using these lowerings in kotlin-native.

**Comment by homuroll** [comment]
Confirm that the PR fixes the [IR VALIDATION] warnings.

File: compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/HeaderProcessor.kt [hunk 1]
```diff
@@ -102,12 +102,12 @@ internal sealed class ForLoopHeader(
                     //   // (use `<` if last is exclusive)
                     //   (step > 0 && inductionVar <= last) || (step < 0 || last <= inductionVar)
                     val stepKotlinType = progressionType.stepType(builtIns).toKotlinType()
-                    val zero = if (progressionType == ProgressionType.LONG_PROGRESSION) irLong(0) else irInt(0)
+                    val isLong = progressionType == ProgressionType.LONG_PROGRESSION;
```
Comment (olonho) [inline compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/HeaderProcessor.kt]: Semicolon may be considered a nostalgic reminiscence here.

Comment (madsager) [reply compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/HeaderProcessor.kt]: Whoops. Old habits die hard. ;-) Removed.

**Comment by max-kammerer** [comment]
Thanks!

**Pull request merged**